bindings/python/tests/test_parse.py

```diff
@@ -32,3 +32,179 @@ def get_parsed_info(tmp_path, file_contents):
     )
 
     return parsed_info
+
+
+def test_struct_decl(tmp_path):
+    file_contents = "struct AStruct {};"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+
+    assert struct_decl["kind"] == "STRUCT_DECL"
+    assert struct_decl["name"] == "AStruct"
+
+
+def test_cxx_base_specifier(tmp_path):
+    file_contents = """
+    struct BaseStruct {};
+    struct DerivedStruct: public BaseStruct {};
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    child_struct_decl = parsed_info["members"][1]
+    cxx_base_specifier = child_struct_decl["members"][0]
+
+    assert cxx_base_specifier["kind"] == "CXX_BASE_SPECIFIER"
+    assert cxx_base_specifier["access_specifier"] == "PUBLIC"
+    assert cxx_base_specifier["name"] == "struct BaseStruct"
+
+
+def test_cxx_method(tmp_path):
+    file_contents = """
+    struct AStruct {
+        void aMethod() {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    cxx_method = struct_decl["members"][0]
+
+    assert cxx_method["kind"] == "CXX_METHOD"
+    assert cxx_method["result_type"] == "void"
+    assert cxx_method["name"] == "aMethod"
+
+
+def test_type_ref(tmp_path):
+    file_contents = """
+    struct AStruct { 
+        int aClassMember;
+        void aMethod(AStruct& aClassMember) {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    cxx_method = struct_decl["members"][1]
+    l_value_ref = cxx_method["members"][0]
+    type_ref = l_value_ref["members"][0]
+
+    assert type_ref["kind"] == "TYPE_REF"
+    assert type_ref["name"] == "struct AStruct"
```

Comment:
From previous review: https://github.com/divmadan/pcl/pull/6#discussion_r459607861

Alternate:
```
    typedef unsigned size_type;
    size_type size;
```

> Is that a type ref? Or is a type-ref a type reference (as in int& and not just int)?

> From cindex:
```
# A reference to a type declaration.
#
# A type reference occurs anywhere where a type is named but not
# declared. For example, given:
#   typedef unsigned size_type;
#   size_type size;
#
# The typedef is a declaration of size_type (CXCursor_TypedefDecl),
# while the type of the variable "size" is referenced. The cursor
# referenced by the type of size is the typedef for size_type.
CursorKind.TYPE_REF = CursorKind(43)
```

Reply:
Maybe don't use reference in the type to avoid confusion

Reply:
Could you please make a suggestion? Thanks.

Reply:
I think he means to say "change line 82 from 
`void aMethod(AStruct& aClassMember) {}` to 
`void aMethod(AStruct aClassMember) {}`. It's referring to a type not referring to a type's reference.

Reply:
Used your suggestion:
> ```
> struct SomeUsefulType {};
> 
> class AClass {
>     void aMethod(SomeUsefulType aParameter) {};
> };
> ```